File: 64062.cpp
using namespace std;

int solution(vector<int> stones, int k) {
    int max = *max_element(stones.begin(), stones.end());
    int min = *min_element(stones.begin(), stones.end());
    int pivot = (max + min) / 2;
    
    while(true) {
        int bestLength = 0;
        int currentLength = 0;
        pivot = (max + min) / 2;
        
        if(min > max) {
            return pivot + 1;
        }
        
        for(int i = 0; i < stones.size(); i++) {
            if(stones[i] - pivot > 0)
                currentLength = 0;
            else {
                currentLength++;
                if(currentLength > bestLength)
                    bestLength = currentLength;
            }
        }
        
        if(bestLength < k) {
            min = pivot + 1;
        } else {
            max = pivot - 1;
        }
    }
}

File: 64062.swift
import Foundation

func solution(_ stones:[Int], _ k:Int) -> Int {
    var from = 0
    var to = 2000000000
    var mid = (from + to) / 2
    
    while from < to {
        var length = 0
        for stone in stones {
            guard length < k else { break }
            if stone - mid <= 0 {
                length += 1
            } else {
                length = 0
            }
        }
        
        if length >= k {
            to = mid
            mid = (from + to) / 2
        } else {
            from = mid + 1
            mid = (from + to) / 2
        }
    }
    
    return mid
}
